frontends/web/src/i18n/i18n.test.tsx

```diff
@@ -57,20 +34,20 @@ describe('i18n', () => {
             {nativeLocale: 'fr', newLang: 'en', userLang: 'en'},
         ];
         table.forEach((test) => {
-            it(`sets userLanguage to ${test.userLang} if native-locale is ${test.nativeLocale}`, (done) => {
-                (apiGet as jest.Mock).mockImplementation(endpoint => {
+            it(`sets userLanguage to ${test.userLang} if native-locale is ${test.nativeLocale}`, async () => {
+                (apiGet as jest.Mock).mockImplementation( endpoint => {
                     switch (endpoint) {
                         case 'config': { return Promise.resolve({}); }
                         case 'native-locale': { return Promise.resolve(test.nativeLocale); }
                         default: { return Promise.resolve(); }
                     }
                 });
-                i18n.listeners.languageChanged(test.newLang).then(() => {
-                    expect(apiPost).toBeCalledWith('config', {
-                        frontend: {},
-                        backend: {userLanguage: test.userLang},
-                    });
-                    done();
+                let callbackPromise = await i18n.changeLanguage(test.newLang);
+                await callbackPromise;
```

Comment:
why does it need 2 `await`? 🤔

Reply:
Because `i18n.changeLanguage` is changing the language of the `i18n`, and doesn't wait it's effects. It returns a prom for them.

Reply:
right. otherwise, `setConfig({backend: ...})` won't have occured yet and `expect(apiPost)` will fail.

would be nice to add a comment so in 3 months from now anyone reading this part understands:

```js
await callbackPromise; // wait for setConfig to complete
```